File: ecommerc_dotnet/core/dto/OrderDto.cs
using System.ComponentModel.DataAnnotations;
using ecommerc_dotnet.dto.Request;
using ecommerc_dotnet.dto.Response;

namespace ecommerc_dotnet.dto
{
    public class OrderDto
    {
        public Guid Id { get; set; }
        public decimal Longitude { get; set; }
        public decimal Latitude { get; set; }
        public decimal TotalPrice { get; set; }
        public decimal DeliveryFee { get; set; }
        public String Name { get; set; }
        public String UserPhone { get; set; }
        public int Status { get; set; }
        public List<OrderItemDto>? OrderItems { get; set; } = null;
    }
    
    public class DeliveryOrderDto
    {
        public Guid Id { get; set; }
        public decimal Longitude { get; set; }
        public decimal Latitude { get; set; }
        public decimal TotalPrice { get; set; }
        public decimal RealPrice { get; set; }
        public decimal DeliveryFee { get; set; }
        public String Name { get; set; }
        public String UserPhone { get; set; }
        public int Status { get; set; }
        public List<DeliveryOrderItemDto> OrderItems { get; set; }
    }

    public class CreateOrderDto
    {
        [Required] public decimal Longitude { get; set; }
        [Required] public decimal Latitude { get; set; }
        [Required] public decimal TotalPrice { get; set; }
        [Required] public List<CreateOrderItemDto> Items { get; set; }
    }

    public class OrderTakedByEvent
    {
        public Guid Id { get; set; }
        public Guid DeliveryId { get; set; }
    }

    public class UpdateOrderStatusDto
    {
        public Guid Id { get; set; }
        public int Status { get; set; } 
    }
}

File: ecommerce-delivery-man/app/src/main/java/com/example/e_commerc_delivery_man/dto/OrderDto.kt
package com.example.eccomerce_app.dto.response

import com.example.hotel_mobile.services.kSerializeChanger.UUIDKserialize
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.util.UUID

@Serializable
data class OrderDto(
    @Serializable(with = UUIDKserialize::class)
    val id: UUID,
    val longitude: Double,
    val latitude: Double,
    val userPhone: String,
    var name: String,
    val status:String,
    val totalPrice: Double,
    val deliveryFee: Double,
    val orderItems:List<OrderItemDto>
)

@Serializable
data class OrderUpdateStatusDto(
    @Serializable(with = UUIDKserialize::class)
    val id: UUID,
    val status:String,
)
@Serializable
data class OrderUpdateEvent(
    @Serializable(with = UUIDKserialize::class)
    val id: UUID,
    @Serializable(with = UUIDKserialize::class)
    val deliveryId: UUID,
)

@Serializable
data class UpdateOrderStatus(
    @Serializable(with = UUIDKserialize::class)
    val Id: UUID,
    @SerialName("Status")
    val Status:Int=5
)
